File: test 7/clojure/test/neural/node_test.clj
(ns neural.node-test
  (:require [clojure.test :refer :all]
            [neural.node :refer :all]))

(deftest test-summate
  (testing "summate"
    (is (= (summate [0.7 3 2.5 4]) 10.2))))

(deftest test-exp
  (testing "exp"
    (is (= 2.7183 (exp)))))

(deftest test-activ-func1
  (testing "activ-func1"
    (is (< (abs (- 1 (activ-func 10))) 0.1))))

(deftest test-activ-func2
  (testing "activ-func2"
    (is (< (abs (- 0 (activ-func -10))) 0.1))))

(deftest test-activ-func3
  (testing "activ-func3"
    (is (< (abs (- 0.5 (activ-func 0))) 0.1))))

File: test 7/typescript/ts/test/node.test.ts
import { it, describe } from "mocha";
import  * as assert from "assert";
import { activFunction } from "../src/node";

describe("test activ function", () => {
    it("test activ function 0.5", () => {
        assert.ok(Math.abs(activFunction(0) - 0.5) < 0.01)
    })
    it("test activ function -10", () => {
        assert.ok(Math.abs(activFunction(-10) - 0) < 0.01)
    })
    it("test activ function 10", () => {
        assert.ok(Math.abs((activFunction(10) - 1)) < 0.01)
    })
})